README.md

```diff
@@ -20,110 +13,1177 @@ visit https://pkg.go.dev/github.com/pinecone-io/go-pinecone/pinecone.
+		}
+	}
+}
+```
+
+### Describe an index
+
+The following example describes an index by name.
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/pinecone-io/go-pinecone/pinecone"
+	"log"
+	"os"
+)
+
+func main() {
+	ctx := context.Background()
+
+	clientParams := pinecone.NewClientParams{
+		ApiKey: os.Getenv("PINECONE_API_KEY"),
+	}
+
+	pc, err := pinecone.NewClient(clientParams)
+	if err != nil {
+		log.Fatalf("Failed to create Client: %v", err)
+	} else {
+		fmt.Println("Successfully created a new Client object!")
 	}
 
-	for _, index := range idxs {
-		fmt.Println(index)
+	indexName := "the-name-of-my-index"
+
+	idx, err := pc.DescribeIndex(ctx, indexName)
+	if err != nil {
+		log.Fatalf("Failed to describe index: %s", err)
+	} else {
+		fmt.Printf("%+v", *idx)
 	}
+}
+```
+
+### Delete an index
+
+The following example deletes an index by name. Note: only indexes not protected by deletion protection
+may be deleted.
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/pinecone-io/go-pinecone/pinecone"
+	"log"
+	"os"
+)
+
+func main() {
+	ctx := context.Background()
 
-	idx, err := pc.Index(idxs[0].Host)
-	defer idx.Close()
+	clientParams := pinecone.NewClientParams{
+		ApiKey: os.Getenv("PINECONE_API_KEY"),
+	}
 
+	pc, err := pinecone.NewClient(clientParams)
 	if err != nil {
-		fmt.Println("Error:", err)
-		return
+		log.Fatalf("Failed to create Client: %v", err)
+	} else {
+		fmt.Println("Successfully created a new Client object!")
 	}
 
-	res, err := idx.DescribeIndexStats(ctx)
+	indexName := "the-name-of-my-index"
+
+	err = pc.DeleteIndex(ctx, indexName)
 	if err != nil {
-		fmt.Println("Error:", err)
-		return
+		log.Fatalf("Error: %v", err)
+	} else {
+		fmt.Printf("Index \"%s\" deleted successfully", indexName)
 	}
+}
+```
 
-	fmt.Println(res)
+### Configure an index
+
+There are multiple ways to configure Pinecone indexes. You are able to configure Deletion Protection for both
+pod-based and Serverless indexes. Additionally, you can configure the size of your pods and the number of replicas
+for pod-based indexes. Examples for each of these configurations are provided below.
+
+```go
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/pinecone-io/go-pinecone/pinecone"
+	"log"
+	"os"
+)
+
+func main() {
+	ctx := context.Background()
+
+	clientParams := pinecone.NewClientParams{
+		ApiKey: os.Getenv("PINECONE_API_KEY"),
+	}
+
+	pc, err := pinecone.NewClient(clientParams)
+	if err != nil {
+		log.Fatalf("Failed to create Client: %v", err)
+	} else {
+		fmt.Println("Successfully created a new Client object!")
+	}
+
+	// To scale the size of your pods from "x2" to "x4" (only applicable to pod-based indexes):
+	_, err := pc.ConfigureIndex(ctx, "my-index", "p1.x4", nil)
+	if err != nil {
+		fmt.Printf("Failed to configure index: %v\n", err)
+	}
+
+	// To scale the number of replicas from 2 to 4 (only applicable to pod-based indexes):
+	_, err := pc.ConfigureIndex(ctx, "my-index", nil, 4)
```

Comment:
Gotta add one more `nil` to these guys (I think) to account for new deletion protection param